packit/upstream.py

```diff
@@ -441,6 +446,26 @@ def get_last_tag(self) -> Optional[str]:
             return None
         return last_tag
 
+    def get_commit_messages(self, after: str = None, before: str = "HEAD") -> str:
+        """
+        :param after: get commit messages after this revision,
+        if None, all commit messages before 'before' will be returned
+        :param before:  get commit messages before this revision
+        :return: commit messages
+        """
+        # let's print changes b/w the last 2 revisions;
+        # ambiguous argument '0.1.0..HEAD': unknown revision or path not in the working tree.
+        # Use '--' to separate paths from revisions, like this
+        commits_range = f"{after}..{before}" if after else before
+        cmd = [
+            "git",
+            "log",
```

Comment:
could you please add here `--no-merges`? I realized it produces much cleaner changelogs:

yes:
```
11:00:25.267 commands.py       DEBUG  Command: git log --no-merges --pretty=format:- %s (%an) 0.17.0..0.18.0 --
11:00:25.273 logging.py        DEBUG  - Update CHANGELOG.md (Jiri Popelka)
11:00:25.273 logging.py        DEBUG  - 0.18.0 release (Release bot)
11:00:25.273 logging.py        DEBUG  - Fix anonymous GitlabService (Matej Focko)
11:00:25.273 logging.py        DEBUG  - Adding delete project functionality (shreyas)
11:00:25.273 logging.py        DEBUG  - Update pre-commit configuration for prettier (Matej Focko)
11:00:25.273 logging.py        DEBUG  - Add assignees argument while creating issue (shreyas)
11:00:26.233 sync.py           DEBUG  Copy synced files...
```

no:
```
10:59:39.008 commands.py       DEBUG  Command: git log --pretty=format:- %s (%an) 0.17.0..0.18.0 --                                                                                           
10:59:39.014 logging.py        DEBUG  - Merge pull request #493 from packit/0.18.0-release (softwarefactory-project-zuul[bot])                                                                
10:59:39.015 logging.py        DEBUG  - Update CHANGELOG.md (Jiri Popelka)                                                                                                                    
10:59:39.015 logging.py        DEBUG  - 0.18.0 release (Release bot)                                                                                                                          
10:59:39.015 logging.py        DEBUG  - Merge pull request #488 from mfocko/gitlab-no-auth (softwarefactory-project-zuul[bot])                                                                
10:59:39.015 logging.py        DEBUG  - Fix anonymous GitlabService (Matej Focko)                                                                                                             
10:59:39.015 logging.py        DEBUG  - Merge pull request #489 from shreyaspapi/issue215 (softwarefactory-project-zuul[bot])                                                                 
10:59:39.015 logging.py        DEBUG  - Adding delete project functionality (shreyas)                                                                                                         
10:59:39.015 logging.py        DEBUG  - Merge pull request #491 from mfocko/fix-prettier (softwarefactory-project-zuul[bot])                                                                  
10:59:39.015 logging.py        DEBUG  - Update pre-commit configuration for prettier (Matej Focko)                                                                                            
10:59:39.015 logging.py        DEBUG  - Merge pull request #487 from shreyaspapi/issue329 (softwarefactory-project-zuul[bot])                                                                 
10:59:39.015 logging.py        DEBUG  - Add assignees argument while creating issue (shreyas)                                                                                                 
10:59:39.962 sync.py           DEBUG  Copy synced files...
```

Reply:
much better :+1: changed